tests/contest/contest/src/tests/exec/cap_test.rs

```diff
@@ -0,0 +1,286 @@
+        }
+
+        TestResult::Passed
+    })
+}
+
+pub(crate) fn get_test_new_privileges() -> TestResult {
+    let no_caps = LinuxCapabilitiesBuilder::default()
+        .bounding(HashSet::new())
+        .effective(HashSet::new())
+        .inheritable(HashSet::new())
+        .permitted(HashSet::new())
+        .ambient(HashSet::new());
+    let spec = test_result!(super::create_spec(Some(
+        ProcessBuilder::default()
+            .no_new_privileges(false)
+            .capabilities(no_caps.build().expect("build no caps failed"))
+    )));
+
+    test_outside_container(&spec, &|data| {
+        test_result!(check_container_created(&data));
+
+        let id = &data.id;
+        let dir = &data.bundle;
+
+        let start_result = start_container(id, dir).unwrap().wait().unwrap();
+        if !start_result.success() {
+            return TestResult::Failed(anyhow!("container start failed"));
+        }
+
+        let (stdout, _) =
+            exec_container(id, dir, &["cat", "/proc/self/status"], None).expect("exec failed");
+
+        if !stdout.contains("CapInh:\t0000000000000000") {
+            return TestResult::Failed(anyhow!("CapInh unexpected output: {}", stdout));
+        }
+        if !stdout.contains("CapAmb:\t0000000000000000") {
+            return TestResult::Failed(anyhow!("CapAmb unexpected output: {}", stdout));
+        }
+        if !stdout.contains("NoNewPrivs:\t0") {
+            return TestResult::Failed(anyhow!("NoNewPrivs unexpected output: {}", stdout));
+        }
+
+        TestResult::Passed
+    })
+}
+
+pub(crate) fn get_test_some_capabilities() -> TestResult {
+    let mut bounding = HashSet::new();
+    bounding.insert(Capability::SysAdmin);
+
+    let mut permitted = HashSet::new();
+    permitted.insert(Capability::SysAdmin);
+    permitted.insert(Capability::AuditWrite);
+    permitted.insert(Capability::Kill);
+    permitted.insert(Capability::NetBindService);
+
+    let caps = LinuxCapabilitiesBuilder::default()
+        .bounding(bounding)
+        .effective(HashSet::new())
+        .inheritable(HashSet::new())
+        .permitted(permitted)
+        .ambient(HashSet::new());
+    let spec = test_result!(super::create_spec(Some(
+        ProcessBuilder::default()
+            .no_new_privileges(true)
+            .capabilities(caps.build().expect("build no caps failed"))
+    )));
+
+    test_outside_container(&spec, &|data| {
+        test_result!(check_container_created(&data));
+
+        let id = &data.id;
+        let dir = &data.bundle;
+
+        let start_result = start_container(id, dir).unwrap().wait().unwrap();
+        if !start_result.success() {
+            return TestResult::Failed(anyhow!("container start failed"));
+        }
+
+        let (stdout, _) =
+            exec_container(id, dir, &["cat", "/proc/self/status"], None).expect("exec failed");
+
+        if !stdout.contains("CapInh:\t0000000000000000") {
```

Comment:
Please add a comment noting that the bounding set limits the capabilities, so only CAP_SYS_ADMIN is actually granted.